ant_plus_sensor/ant_plus_logger.js

```diff
@@ -160,20 +160,10 @@ async function heartRateConnect(antPlus) {
     if (isRecording) {
       const power = Math.round(powerAverage.average() * 100) / 100;
       const payload = {
-        'module-id': moduleID,
         sensors: [
-          {
-            type: 'power',
-            value: power,
-          },
-          {
-            type: 'cadence',
-            value: cadence,
-          },
-          {
-            type: 'heartRate',
-            value: heartRate,
-          },
+          ...(power ? [{ type: 'power', value: power }] : []),
+          ...(cadence ? [{ type: 'cadence', value: cadence }] : []),
+          ...(heartRate ? [{ type: 'heartRate', value: heartRate }] : []),
```

Comment:
Does dashboard handle an empty array for all these types?

Reply:
Because of the `...` these arrays are being destructured into the parent array. So, if these fields are null, their entry won't be included in `sensors` array at all.

Think of it as either appending the array `[{ type: 'power', value: power }]` if `power` is not null, else `[]` to the sensor array.